httputil/handler.go

```diff
@@ -0,0 +1,69 @@
+package httputil
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+)
+
+// ErrorHandler factors error handling out of the HTTP handler.
+type ErrorHandler struct {
+	Handler func(w http.ResponseWriter, req *http.Request) error
```

Comment:
That's a good idea, thanks for the suggestion Ian! It's especially nice because it would avoid the need for much of the unpleasant change in d5c4867a91a00291ae52d73bfc3ed36172a04272.

I think I made it a struct originally because I wasn't sure if there'd be 1 or more fields, but by now it's becoming increasingly clear that it will stay as the one func field.